vars/transportRequestCreate.groovy

```diff
@@ -0,0 +1,68 @@
+import com.sap.piper.GitUtils
+import groovy.transform.Field
+
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.cm.ChangeManagement
+import com.sap.piper.cm.ChangeManagementException
+
+import hudson.AbortException
+
+
+@Field def STEP_NAME = 'transportRequestCreate'
+
+@Field Set parameterKeys = [
+    'changeId',
+    'developmentSystemId',
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+@Field Set stepConfigurationKeys = [
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+def call(parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        ChangeManagement cm = new ChangeManagement(script)
+
+        Map configuration = ConfigurationMerger.merge(parameters.script, STEP_NAME,
+                                                      parameters, parameterKeys,
+                                                      stepConfigurationKeys)
+
+        def changeId = configuration.changeId
+        if(!changeId) throw new AbortException('Change id not provided (parameter: \'changeId\').')
+
+        def developmentSystemId = configuration.developmentSystemId
+        if(!developmentSystemId) throw new AbortException('Development system id not provided (parameter: \'developmentSystemId\').')
+
+        def cmCredentialsId = configuration.cmCredentialsId
+        if(!cmCredentialsId) throw new AbortException('Credentials id not provided (parameter: \'cmCredentialsId\').')
+
+        def cmEndpoint = configuration.cmEndpoint
+        if(!cmEndpoint) throw new AbortException('Solution Manager endpoint not provided (parameter: \'cmEndpoint\').')
+
+        def transportRequestId
+
+        echo "[INFO] Creating transport request for change document '$changeId' and development system '$developmentSystemId'."
+
+        withCredentials([usernamePassword(
```

Comment:
Is there any reason for having the withCredentials not inside the ```ChangeManagment```class? Just asking??? Is that method not accessible from the class, but is accessible here?

Reply:
here can be accessed directly. As the `ChangeManagment` class has a reference to the script I guess it can also be called there.

Reply:
Reason: limit the code block knowing the credentials to the bare minimum. But OK, if that is difficult ...

Reply:
Can this be done when you introduce the CM_OPTS?

Reply:
Can be done later, either before or after merging this or another pull request.